### PR TITLE
Flush out access transformer grammars to handle more cases

### PR DESCRIPTION
**Pull request by DemonWav**
Convert all AT code to Kotlin

### CODE REVIEW

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/AtLanguage.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.mcp.at
+
+import com.intellij.lang.Language
+
+object AtLanguage : Language("Access Transformers")
```
Comment (Minecrell) [inline src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/AtLanguage.kt]: :(

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/AtMixinUtil.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        newText = newText.replace("\\[".toRegex(), "")
+
+        if (newText.startsWith("L")) {
+            // class
+            newText = newText.substring(1)
+
+            newText = newText.replace(";".toRegex(), "")
+            newText = newText.replace("/".toRegex(), ".")
+
+            return instance.findClass(newText, GlobalSearchScope.allScope(project))
```
Comment (Minecrell) [inline src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/AtMixinUtil.kt]: You need to use `findQualifiedClass` if you want to handle inner classes.

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/AtGotoDeclarationHandler.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        val module = ModuleUtilCore.findModuleForPsiElement(sourceElement) ?: return null
+
+        val instance = MinecraftFacet.getInstance(module) ?: return null
+
+        val mcpModule = instance.getModuleOfType(McpModuleType) ?: return null
+
+        val srgMap = mcpModule.srgManager.srgMapNow ?: return null
+
+        if (sourceElement.node.treeParent.elementType === AtTypes.CLASS_NAME) {
```
Comment (Minecrell) [inline src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/AtGotoDeclarationHandler.kt]: This looks like a `when`

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/AtMixinUtil.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    newText = newText.replace("\\[".toRegex(), "")
+
+    val scope = GlobalSearchScope.allScope(project)
+
+    if (newText.startsWith("L")) {
+        // class
+        newText = newText.substring(1)
+
+        newText = newText.replace(";", "")
+        newText = newText.replace("/", ".")
```
Comment (Minecrell) [inline src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/AtMixinUtil.kt]: There is `PsiBytecodeUtil.parseClassDescriptor` which does exactly this, doesn't it?

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/AtMixinUtil.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+        return findQualifiedClass(project, newText, scope)
+    }
+
+    val instance = JavaPsiFacadeEx.getInstanceEx(project)
+
+    instance.apply {
+        // Primitive
+        when (newText) {
+            "B" -> return findClass(Int::class.java.canonicalName, scope)
```
Comment (Minecrell) [inline src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/AtMixinUtil.kt]: You could use `PsiBytecodeUtil.getPrimitiveType(newText).getBoxedType` here for some reduced code duplication

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/AtGotoDeclarationHandler.kt (hunk 3 above)
Comment (DemonWav) [inline src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/AtGotoDeclarationHandler.kt]: Not quite, not all the branches are comparing the same thing.

Comment (Minecrell) [inline src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/AtGotoDeclarationHandler.kt]:
```kt
when (a) {
    A -> {}
    B -> when (b) {
        A -> {}
        B -> when (c) {
            A -> {}
            B -> {}
        }
    }
}
```

**Pull request merged**